### DISCUSSION TITLE
How to correctly initialize GetXController in tabpage?

### DISCUSSION THREAD

**Discussion by princ3od** [jonataslaw/getx#1492] [Q&A] [answered]
My HomePage has a PageView which contains 4 pages (tabs) and combines with BottomNavigationBar.
```Dart
PageView(
              physics: BouncingScrollPhysics(),
              onPageChanged: controller.onPageChanged,
              controller: controller.pageController,
              children: [
                HomeTab(),
                LearningTab(),
                TeachingTab(),
                NotificationTab(),
              ],
            ),
```
Each tab has its own Controller like this:
```Dart
class LearningTab extends StatelessWidget {
  final LearningController controller =
      Get.put<LearningController>(LearningController());
  @override
  Widget build(BuildContext context) {
    return Stuff(...);
  }
}
```
The problem is when HomePage loads, all the controllers of 4 tabs also initialize at the same time. 
The performance is still fine but i think this behavior doesn't look right.
How to make only controller of selected tab intialize and when switch to another tab, the controller of previous tab is deleted?

**Comment by ad-on-is** [comment, marked as answer]
This behavior definitely **looks right**... Imagine your PageController() had a viewportFraction, where you can see a fraction of other pages. What would these fractions show, if these FooTabs() wouldn't be fully initialized.
